packages/jest-runtime/src/index.ts

```diff
@@ -664,14 +667,24 @@ class Runtime {
       localModule.exports = this._environment.global.JSON.parse(
         transformedFile,
       );
-    } else if (path.extname(modulePath) === '.node') {
+    } else if (
+      path.extname(modulePath) === '.node' &&
+      moduleSource === undefined
```

Comment:
Just as a note for the future when finalizing this: This check being there should probably be ensured by a test case.